internal/step/snapshot.go

```diff
@@ -16,8 +16,10 @@ import (
 // Snapshot creates a moment in time record of the
 // step and attempts to upload it to the server.
 func Snapshot(ctn *pipeline.Container, b *library.Build, c *vela.Client, l *logrus.Entry, r *library.Repo, s *library.Step) {
-	// check if the build is not in a canceled status
-	if !strings.EqualFold(s.GetStatus(), constants.StatusCanceled) {
+	// check if the build is not in a canceled status or error status
+	logrus.Debugf("Snapshot s: %s %s", s.GetName(), s.GetStatus())
+	if !strings.EqualFold(s.GetStatus(), constants.StatusCanceled) &&
+		!strings.EqualFold(s.GetStatus(), constants.StatusError) {
```

Comment:
Was this an existing oversight?

Reply:
Yes. I noticed this when the container was in error status and got set to success status here.